src/main/java/edu/ie3/datamodel/utils/ContainerUtils.java

```diff
@@ -131,6 +140,149 @@ private static void addDistanceGraphEdge(
+   */
+  private static void addImpedanceGraphEdge(
+      ImpedanceWeightedGraph graph, ConnectorInput connectorInput) {
+    NodeInput nodeA = connectorInput.getNodeA();
+    NodeInput nodeB = connectorInput.getNodeB();
+    /* Add an edge if it is not a switch or the switch is closed */
+    if (!(connectorInput instanceof SwitchInput) || ((SwitchInput) connectorInput).isClosed())
+      graph.addEdge(nodeA, nodeB);
+
+    if (connectorInput instanceof LineInput) {
+      LineInput line = (LineInput) connectorInput;
+      graph.setEdgeWeight(
+          graph.getEdge(nodeA, nodeB),
+          calcImpedance(line.getType().getR(), line.getType().getX(), line.getLength()));
+    }
+    if (connectorInput instanceof SwitchInput) {
+      SwitchInput sw = (SwitchInput) connectorInput;
+      // assumption: closed switch has a resistance of 1 OHM
+      if (sw.isClosed()) graph.setEdgeWeight(graph.getEdge(nodeA, nodeB), 1);
+    }
+    if (connectorInput instanceof Transformer2WInput) {
+      Transformer2WInput trafo2w = (Transformer2WInput) connectorInput;
+      graph.setEdgeWeight(
+          graph.getEdge(nodeA, nodeB),
+          sqrt(
+              pow(trafo2w.getType().getrSc().to(OHM).getValue().doubleValue(), 2)
+                  + pow(trafo2w.getType().getxSc().to(OHM).getValue().doubleValue(), 2)));
+    }
+    if (connectorInput instanceof Transformer3WInput) {
+      Transformer3WInput trafo3w = (Transformer3WInput) connectorInput;
+      graph.addEdge(nodeA, trafo3w.getNodeC());
+
+      graph.setEdgeWeight(
+          graph.getEdge(nodeA, nodeB),
+          sqrt(
+              pow(
+                      trafo3w.getType().getrScA().to(OHM).getValue().doubleValue()
+                          + trafo3w.getType().getrScB().to(OHM).getValue().doubleValue(),
+                      2)
+                  + pow(
+                      trafo3w.getType().getxScA().to(OHM).getValue().doubleValue()
+                          + trafo3w.getType().getxScB().to(OHM).getValue().doubleValue(),
+                      2)));
+
+      graph.setEdgeWeight(
+          graph.getEdge(nodeA, trafo3w.getNodeC()),
+          sqrt(
+              pow(
+                      trafo3w.getType().getrScA().to(OHM).getValue().doubleValue()
+                          + trafo3w.getType().getrScC().to(OHM).getValue().doubleValue(),
+                      2)
+                  + pow(
+                      trafo3w.getType().getxScA().to(OHM).getValue().doubleValue()
+                          + trafo3w.getType().getxScC().to(OHM).getValue().doubleValue(),
+                      2)));
+    }
+  }
+
+  private static double calcImpedance(
+      ComparableQuantity<SpecificResistance> r,
+      ComparableQuantity<SpecificResistance> x,
+      ComparableQuantity<Length> length) {
+    return sqrt(
+            pow(r.to(OHM_PER_KILOMETRE).getValue().doubleValue(), 2)
+                + pow(x.to(OHM_PER_KILOMETRE).getValue().doubleValue(), 2))
```

Comment:
If you implement the general calc impedance method you can reference it here as well